api/routes/games.js

```diff
@@ -0,0 +1,245 @@
+        res.status(201)
+          .set({ 
+            "Content-Type": "application/json",
+            "Content-Location": game.url
+          })
+          .send(game);
+      });
+    } 
+
+    // If no categories, send game object with empty category array
+    else {
+      game.categories = [];
+      res.status(201)
+        .set({ 
+          "Content-Type": "application/json",
+          "Content-Location": game.url
+        })
+        .send(game);
+    }
+
+  });
+
+});
+
+/* Upload game image to S3 */
+router.post('/sign-s3', (req, res) => {
+  const S3 = new AWS.S3();
+  const fileName = Date.now() + '-' + req.body.fileName;
+  const fileType = req.body.fileType;
+
+  const s3Params = {
+    Bucket: S3_BUCKET,
+    Key: 'game/' + fileName,
+    ContentType: fileType,
+    ACL: 'public-read',
+  };
+
+  S3.getSignedUrl('putObject', s3Params, (err, data) => {
+    if(err){
+      console.log(err);
+      return res.end();
+    }
+    const returnData = {
+      signedRequest: data,
+      url: `https://${S3_BUCKET}.s3.amazonaws.com/${fileName}`
+    };
+    res.json({ data: {returnData} });
+  });
+});
+
+/* Get reviews for a game */
+router.get('/:game_id/reviews', (req, res) => {
+  let hostname = req.protocol + '://' + req.headers.host;
+
+  let query = {
+    text: 'SELECT * FROM "Review" WHERE "gameID" = $1',
+    values: [req.params.game_id]
+  }
+
+  db.client.query(query, (err, result) => {
+    if (err) {
+      return res.status(400).send(err);
+    }
+
+    let reviews = {};
+    reviews.results = result.rows.map(e => apiUtils.formatReview(e, hostname));
+
+    res.status(200)
+      .set({ "Content-Type": "application/json" })
+      .send(reviews);
+  });
+});
+
+/* Create review for game */
+router.post('/:game_id/reviews', (req, res) => {
+  let hostname = req.protocol + '://' + req.headers.host;
+
+  // Map form field names to database field names
+  const fields = {
+    overallRating: '"overallRating"',
+    comments: 'comments',
+    strategy: '"strategy"',
+    luck: '"luck"',
+    playerInteraction: '"playerInteraction"',
+    replayValue: '"replayValue"',
+    complexity: '"complexity"',
+    gfKids: '"gfKids"',
+    gfTeens: '"gfTeens"',
+    gfAdults: '"gfAdults"',
+    gfFamilies: '"gfFamilies"',
+    gf2Player: '"gf2Player"',
+    gfLargeGroups: '"gfLargeGroups"',
+    gfSocialDistancing: '"gfSocialDistancing"'
+  };
+
+  // Create array of field names for query
+  let query_fields = Object.keys(fields).map(e => fields[e]);
+
+  // Build query object
+  let query = { text: '', values: [] };
+
+  // Add rating values
+  for (let field in fields) {
+    let val;
+
+    if (field === 'comments') {
+      val = req.body[field] || null;
+    } else if (field.startsWith('gf')) {
+      val = req.body[field] === 'true';
+    } else {
+      val = parseInt(req.body[field]) || null;
+    }
+
+    query.values.push(val);
+  }
+
+  // TODO: Add user details
+  query_fields.push('"userID"');
+  query.values.push(1);
```

Comment:
Just for my understanding. This is hardcoding a user right now, with the intention of making it dynamic later, right?

Reply:
Correct